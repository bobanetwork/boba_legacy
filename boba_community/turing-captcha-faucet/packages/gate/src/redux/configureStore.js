import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';

import reducers from './reducers';
import initialState from './initialState';

const store = createStore(reducers, initialState, applyMiddleware(thunk));

export default store;
