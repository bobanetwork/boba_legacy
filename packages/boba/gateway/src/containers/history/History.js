/*
Copyright 2019-present OmiseGO Pte Ltd

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

     http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License. */

import React, { useState } from 'react'
import { batch, useDispatch } from 'react-redux'
import { isEqual, orderBy } from 'lodash'
import { useSelector } from 'react-redux'
import DatePicker from 'react-datepicker'
import "react-datepicker/dist/react-datepicker.css"

import { useMediaQuery, useTheme } from '@mui/material'

import moment from 'moment'
import Input from 'components/input/Input'

import { setActiveHistoryTab } from 'actions/uiAction'
import { selectActiveHistoryTab } from 'selectors/uiSelector'

import { fetchTransactions } from 'actions/networkAction'
import { selectTransactions } from 'selectors/transactionSelector'
import { selectLayer } from 'selectors/setupSelector'

import Exits from './TX_Exits'
import Deposits from './TX_Deposits'
import All from './TX_All'
import Pending from './TX_Pending'

import * as S from './History.styles'
import * as styles from './TX_All.module.scss'

import useInterval from 'util/useInterval'
import PageTitle from 'components/pageTitle/PageTitle'
import Connect from 'containers/connect/Connect'
import Tabs from 'components/tabs/Tabs'

import { POLL_INTERVAL } from 'util/constant'

function History() {

  const theme = useTheme()
  const isMobile = useMediaQuery(theme.breakpoints.down('md'))

  const dispatch = useDispatch()

  const now = new Date()
  const last_6months = new Date(now.getFullYear(), now.getMonth()-6, now.getDate())

  const [startDate, setStartDate] = useState(last_6months)
  const [endDate, setEndDate] = useState(now)
  const layer = useSelector(selectLayer())

  const [searchHistory, setSearchHistory] = useState('')
  const activeTab = useSelector(selectActiveHistoryTab, isEqual)

  const unorderedTransactions = useSelector(selectTransactions, isEqual)
  const orderedTransactions = orderBy(unorderedTransactions, i => i.timeStamp, 'desc')

  const transactions = orderedTransactions.filter((i) => {
    if (startDate && endDate) {
      return (moment.unix(i.timeStamp).isSameOrAfter(startDate) && moment.unix(i.timeStamp).isSameOrBefore(endDate))
    }
    return true
  })

  useInterval(() => {
    batch(()=>{
      dispatch(fetchTransactions())
    })
  }, POLL_INTERVAL)

  return (

    <S.HistoryPageContainer>

      <PageTitle title={'History'} />
      
      <Connect 
        userPrompt={'Connect to MetaMask to see your history'}
        accountEnabled={layer}
      />

      {layer && <>
      <S.Header>
        <div className={styles.searchInput}>
          <Input
            size='small'
            placeholder='Search by hash'
            value={searchHistory}
            onChange={i=>{setSearchHistory(i.target.value)}}
            className={styles.searchBar}
          />
        </div>
        <div className={styles.actions}>
          {!isMobile ? (
            <div style={{ margin: '0px 10px', opacity: 0.7 }}>Show period from</div>
          ) : null}
          <DatePicker
            wrapperClassName={styles.datePickerInput}
            popperClassName={styles.popperStyle}
            selected={startDate}
            onChange={(date)=>setStartDate(date)}
            selectsStart
            endDate={new Date(endDate)}
            maxDate={new Date(endDate)}
            calendarClassName={theme.palette.mode}
            placeholderText={isMobile ? "From" : ""}
          />
          {!isMobile ? (
            <div style={{ margin: '0px 10px', opacity: 0.7 }}>to </div>
          ) : null}
          <DatePicker
            wrapperClassName={styles.datePickerInput}
            popperClassName={styles.popperStyle}
            selected={endDate}
            onChange={(date)=>setEndDate(date)}
            selectsEnd
            startDate={new Date(startDate)}
            minDate={new Date(startDate)}
            calendarClassName={theme.palette.mode}
            placeholderText={isMobile ? "To" : ""}
          />
        </div>
      </S.Header>
      <div className={styles.data}>
        <div className={styles.section}>
          <Tabs
            onClick={tab => {dispatch(setActiveHistoryTab(tab))}}
            activeTab={activeTab}
            tabs={['All', 'Bridge to L2', 'Bridge to L1', 'Pending']}
          />

          {activeTab === 'All' && (
            <All
              searchHistory={searchHistory}
              transactions={transactions}
            />
          )}

          {activeTab === 'Bridge to L2' &&
            <Deposits
              searchHistory={searchHistory}
              transactions={transactions}
            />
          }

          {activeTab === 'Bridge to L1' &&
            <Exits
              searchHistory={searchHistory}
              transactions={transactions}
            />
          }

          {activeTab === 'Pending' &&
            <Pending
              searchHistory={searchHistory}
              transactions={transactions}
            />
          }
        </div>
      </div>
    </>}
    </S.HistoryPageContainer>
  );
}

export default React.memo(History)
