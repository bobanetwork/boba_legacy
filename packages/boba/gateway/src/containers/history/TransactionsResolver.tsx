import { TransactionsTableContent } from 'components/global/table/themes'
import dayjs from 'dayjs'
import React, { useState, useEffect } from 'react'
import { getCoinImage } from 'util/coinImage'
import { formatDate, isSameOrAfterDate, isSameOrBeforeDate } from 'util/dates'
import { ALL_NETWORKS, Chains } from './constants'
import { Svg } from 'components/global/svg'
import { TokenInfo } from 'containers/history/tokenInfo'

import {
  TransactionDate,
  TransactionAmount,
  Icon,
  Status,
  NoHistory,
  TransactionDetails,
  TransactionChainDetails,
  TransactionToken,
  TransactionsWrapper,
  TransactionHash,
  TransactionChain,
  IconContainer,
  Image,
  IncompleteTransactionHash,
} from './styles'
import {
  ITransactionsResolverProps,
  ITransaction,
  IProcessedTransaction,
  TRANSACTION_STATUS,
  TRANSACTION_FILTER_STATUS,
  LAYER,
} from './types'
import { orderBy } from 'util/lodash'
import truncate from 'truncate-middle'
import { logAmount } from 'util/amountConvert'
import noHistoryIcon from 'assets/images/noHistory.svg'
import bobaLogo from 'assets/images/Boba_Logo_White_Circle.png'

export const TransactionsResolver: React.FC<ITransactionsResolverProps> = ({
  transactions,
  transactionsFilter,
  loading = false,
}) => {
  const [currentTransactions, setCurrentTransactions] = useState<
    ITransaction[]
  >([])

  useEffect(() => {
    setCurrentTransactions(transactions)
  }, [transactions])

  const orderedTransactions = orderBy(
    currentTransactions,
    (i) => i.timeStamp,
    'desc'
  )

  const dateFilter = (transaction: ITransaction) => {
    const txnAfterStartDate = transactionsFilter.startDate
      ? isSameOrAfterDate(
          transaction.timeStamp,
          dayjs(transactionsFilter.startDate)
        )
      : true
    const txnBeforeEndDate = transactionsFilter.endDate
      ? isSameOrBeforeDate(
          transaction.timeStamp,
          dayjs(transactionsFilter.endDate)
        )
      : true

    return txnAfterStartDate && txnBeforeEndDate
  }

  // should filter out transctions that aren't cross domain
  const crossDomainFilter = (transaction: ITransaction) => {
    return (
      transaction.crossDomainMessage &&
      transaction.crossDomainMessage.crossDomainMessage
    )
  }

  const networkFilter = (transaction: ITransaction) => {
    let targetFromNetwork = false
    let targetToNetwork = false
    switch (transactionsFilter.fromNetworkChainId) {
      case ALL_NETWORKS.value: {
        targetFromNetwork = true
        break
      }
      case transaction.originChainId.toString(): {
        targetFromNetwork = true
        break
      }
    }
    switch (transactionsFilter.toNetworkChainId) {
      case ALL_NETWORKS.value: {
        targetToNetwork = true
        break
      }
      case transaction.destinationChainId.toString(): {
        targetToNetwork = true
        break
      }
    }
    return targetFromNetwork && targetToNetwork
  }
  const getTransactionStatus = (transaction: ITransaction) => {
    if (
      transaction.action &&
      transaction.crossDomainMessage.crossDomainMessageSendTime
    ) {
      switch (transaction.action.status) {
        case TRANSACTION_STATUS.Succeeded: {
          return TRANSACTION_FILTER_STATUS.Completed
        }
        case TRANSACTION_STATUS.Pending: {
          return TRANSACTION_FILTER_STATUS.Pending
        }
      }
    }
    return TRANSACTION_FILTER_STATUS.Canceled
  }
  const statusFilter = (transaction: ITransaction) => {
    const status = getTransactionStatus(transaction)
    transaction.UserFacingStatus = status
    if (
      transactionsFilter.status &&
      transactionsFilter.status !== TRANSACTION_FILTER_STATUS.All
    ) {
      return transactionsFilter.status === transaction.UserFacingStatus
    }

    return true
  }

  // filter out transactions whose hash does not match
  const hashFilter = (transaction: IProcessedTransaction) => {
    if (transactionsFilter.targetHash) {
      if (
        !transaction.fromHash.includes(transactionsFilter.targetHash) &&
        !transaction.toHash.includes(transactionsFilter.targetHash)
      ) {
        return false
      }
    }
    return true
  }
  const filteredTransactions = orderedTransactions.filter((transaction) => {
    return (
      crossDomainFilter(transaction) &&
      networkFilter(transaction) &&
      statusFilter(transaction) &&
      dateFilter(transaction)
    )
  })

  const process_transaction = (transaction: ITransaction) => {
    let amountString = ''
    const chain = transaction.layer === 'L1pending' ? 'L1' : transaction.layer

    // TODO: have a unknown token to use
    let token = {
      name: 'Etheruem',
      symbol: 'ETH',
      decimals: 18,
    }
    if (
      TokenInfo[transaction.originChainId.toString()][
        transaction.action.token.toLowerCase()
      ]
    ) {
      token =
        TokenInfo[transaction.originChainId.toString()][
          transaction.action.token.toLowerCase()
        ]
    }

    const symbol = token.symbol

    amountString = logAmount(transaction.action.amount, token.decimals, 4)
    const fromHash = transaction.hash
    let toHash = ''
    if (chain === LAYER.L2 && transaction.crossDomainMessage.l1Hash) {
      toHash = transaction.crossDomainMessage.l1Hash
    } else if (chain === LAYER.L1 && transaction.crossDomainMessage.l2Hash) {
      toHash = transaction.crossDomainMessage.l2Hash
    }
    const processedTransaction: IProcessedTransaction = {
      timeStamp: transaction.timeStamp,
      from: transaction.from,
      fromHash,
      toHash,
      to: transaction.to,
      tokenSymbol: symbol,
      amount: amountString,
      status: transaction.UserFacingStatus,
      originChainId: transaction.originChainId,
      destinationChainId: transaction.destinationChainId,
    }
    return processedTransaction
  }

  const processedTransactions = filteredTransactions.map((transaction) => {
    return process_transaction(transaction)
  })

  const filteredProcessedTransactions = processedTransactions.filter(
    (transaction) => {
      return hashFilter(transaction)
    }
  )

  const getTransactionToken = (symbol: string) => {
    return (
      <TransactionToken>
        <IconContainer>
          {symbol === 'BOBA' ? (
            <Image src={bobaLogo} alt="boba network" />
          ) : (
            <Icon src={getCoinImage(symbol)} />
          )}
        </IconContainer>
        <div>{symbol}</div>
      </TransactionToken>
    )
  }

  const getTransactionChain = (
    chainID: string,
    hash: string,
    status: string
  ) => {
    const linkToHash = `${Chains[chainID].transactionUrlPrefix}${hash}`
    const networkName = Chains[chainID].name
    const imgSrc = Chains[chainID].imgSrc

    return (
      <TransactionDetails>
        <IconContainer>
          <Icon src={imgSrc} />
        </IconContainer>
        <TransactionChainDetails>
          <TransactionChain>{networkName}</TransactionChain>
          {hash && (
            <TransactionHash
              href={linkToHash}
              target={'_blank'}
              rel="noopener noreferrer"
            >
              {`Tx: ${truncate(hash, 4, 4, '...')}`}
            </TransactionHash>
          )}
          {!hash && (
            <IncompleteTransactionHash>{`Tx: ${status}`}</IncompleteTransactionHash>
          )}
        </TransactionChainDetails>
      </TransactionDetails>
    )
  }

  const getTransactionDate = (timeStamp: number) => {
    return (
      <TransactionDate>
        {formatDate(timeStamp, 'DD MMM YYYY hh:mm A')}
      </TransactionDate>
    )
  }

  const getTransactionAmount = (amount: string) => {
    return amount ? (
      <TransactionAmount>{amount}</TransactionAmount>
    ) : (
      <TransactionAmount>Not Available</TransactionAmount>
    )
  }
  return (
    <>
      {transactions.length === 0 && (
        <NoHistory
          style={{ marginLeft: 'auto', marginRight: 'auto', padding: '20px' }}
        >
          <Svg src={noHistoryIcon} />
          <div>Transactions Loading...</div>
        </NoHistory>
      )}
      {filteredProcessedTransactions && (
        <TransactionsWrapper>
          {filteredProcessedTransactions.map(
            (transaction: IProcessedTransaction, index) => {
              return (
                <TransactionsTableContent
                  key={`transaction-${index}`}
                  options={[
                    {
                      content: getTransactionDate(transaction.timeStamp),
                      width: 168,
                    },
                    {
                      content: getTransactionChain(
                        transaction.originChainId.toString(),
                        transaction.fromHash,
                        transaction.status
                      ),
                      width: 142,
                    },
                    {
                      content: getTransactionChain(
                        transaction.destinationChainId.toString(),
                        transaction.toHash,
                        transaction.status
                      ),
                      width: 142,
                    },
                    {
                      content: getTransactionToken(transaction.tokenSymbol),
                      width: 90,
                    },
                    {
                      content: getTransactionAmount(transaction.amount),
                      width: 80,
                    },
                    {
                      content: <Status>{transaction.status}</Status>,
                      width: 88,
                    },
                  ]}
                />
              )
            }
          )}
        </TransactionsWrapper>
      )}

      {filteredProcessedTransactions.length === 0 &&
        transactions.length !== 0 &&
        !loading && (
          <NoHistory
            style={{ marginLeft: 'auto', marginRight: 'auto', padding: '20px' }}
          >
            <Svg src={noHistoryIcon} />
            <div>No Transactions Matching Filter.</div>
          </NoHistory>
        )}
    </>
  )
}
