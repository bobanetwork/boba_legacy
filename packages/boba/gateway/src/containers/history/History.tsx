/*
Copyright 2021-present Boba Network.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

     http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License. */

import React, { useState } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { isEqual } from 'util/lodash'
import { ValidValuesFromArray } from 'util/objectManipulation'

import { useTheme } from 'styled-components'

import { Button } from 'components/global'

import transctionService from 'services/transaction.service'
import { NETWORK_TYPE } from 'util/network/network.util'
import {
  ALL_NETWORKS,
  FILTER_OPTIONS,
  NETWORK_L1_OPTIONS,
  NETWORK_L2_OPTIONS,
  TableOptions,
} from './constants'

import {
  selectAccountEnabled,
  selectLayer,
  selectTransactions,
} from 'selectors'

import { fetchTransactions } from 'actions/networkAction'

import {
  Table,
  NoHistory,
  HistoryPageContainer,
  TableHeader,
  TableFilters,
  NetworkDropdowns,
  DateDescriptions,
  SwitchChainIcon,
  SwitchIcon,
  TableTransactionsContainer,
  DatePickerWrapper,
  DropdownNetwork,
  MobileDateDescriptions,
  MobileDatePickerWrapper,
} from './styles'

import { setConnect } from 'actions/setupAction'
import useInterval from 'hooks/useInterval'

import { POLL_INTERVAL } from 'util/constant'

import FilterIcon from 'assets/images/filter.svg'
import noHistoryIcon from 'assets/images/noHistory.svg'
import { FilterDropDown } from 'components/filter'
import { Svg } from 'components/global/svg'
import { TransactionsTableHeader } from 'components/global/table/themes'
import { TransactionsResolver } from './TransactionsResolver'
import { CHAIN_NAME, TRANSACTION_FILTER_STATUS } from './types'
import { SearchInput } from 'components/global/searchInput'

import { Typography } from 'components/global/typography'
import DatePicker from './DatePicker'

const History = () => {
  const [toNetwork, setToNetwork] = useState(ALL_NETWORKS)
  const [fromNetwork, setFromNetwork] = useState(ALL_NETWORKS)
  const [transactionsFound, setTransactionsFound] = useState(true)
  const [switched, setSwitched] = useState(false)

  const theme: any = useTheme()

  const dispatch = useDispatch<any>()

  const now = new Date()
  const last_6months = new Date(
    now.getFullYear(),
    now.getMonth() - 6,
    now.getDate()
  )
  const handleSwitchDropdowns = () => {
    const temp = fromNetwork
    setFromNetwork(toNetwork)
    setToNetwork(temp)
    setSwitched((current) => !current)
  }

  const [filterStartDate, setFilterStartDate] = useState(last_6months)
  const [transactionStatus, setTransactionStatus] = useState('All')
  const [filterEndDate, setFilterEndDate] = useState(now)
  const layer = useSelector(selectLayer())
  const accountEnabled = useSelector(selectAccountEnabled())

  const [searchHistory, setSearchHistory] = useState('')

  const transactions = useSelector(selectTransactions, isEqual)

  // TODO: not working as implementation needs be rewrite.
  const getDatePicker = (label: string, range: boolean = false) => {
    const dateSelector = (date: Date) => {
      label === 'To' ? setFilterEndDate(date) : setFilterStartDate(date)
    }
    return (
      <DatePicker
        selected={label === 'To' ? filterEndDate : filterStartDate}
        onChange={(date: Date) =>
          date && !Array.isArray(date) && dateSelector(date)
        }
        timeFormat="MM/DD/YYYY"
        range={range}
        {...(range
          ? { onChangeFrom: setFilterStartDate, onChangeTo: setFilterEndDate }
          : {})}
        {...(label === 'To'
          ? { minDate: filterStartDate }
          : { maxDate: filterEndDate })}
      />
    )
  }
  const syncTransactions = async () => {
    if (accountEnabled) {
      const newTransactions = await transctionService.getTransactions()
      if (newTransactions.length === 0) {
        setTransactionsFound(false)
      } else {
        setTransactionsFound(true)
      }
      if (
        new Set(ValidValuesFromArray(transactions)).size !==
        new Set(newTransactions).size
      ) {
        dispatch(fetchTransactions())
      }
    }
  }

  useInterval(async () => {
    await syncTransactions()
  }, POLL_INTERVAL)

  return (
    <HistoryPageContainer>
      {layer && (
        <>
          <TableHeader>
            <SearchInput
              placeholder="Search Here"
              value={searchHistory}
              onChange={(i: any) => {
                setSearchHistory(i.target.value)
              }}
            />
            <DatePickerWrapper>
              <DateDescriptions variant="body2">
                Date Range From
              </DateDescriptions>
              {getDatePicker('From')}
              <DateDescriptions variant="body2">To</DateDescriptions>
              {getDatePicker('To')}
            </DatePickerWrapper>
            <MobileDatePickerWrapper>
              <MobileDateDescriptions variant="body2">
                Date Range
              </MobileDateDescriptions>
              {getDatePicker('', true)}
            </MobileDatePickerWrapper>
          </TableHeader>

          <Table>
            <div
              style={{
                width: '100%',
                display: 'flex',
                flexDirection: 'column',
              }}
            >
              <TableFilters
                style={{
                  width: '100%',
                }}
              >
                <NetworkDropdowns>
                  <Typography variant="body2">From</Typography>
                  <DropdownNetwork
                    items={switched ? NETWORK_L2_OPTIONS : NETWORK_L1_OPTIONS}
                    defaultItem={fromNetwork}
                    onItemSelected={(option) => setFromNetwork(option)}
                    error={false}
                    headers={[NETWORK_TYPE.MAINNET, NETWORK_TYPE.TESTNET]}
                  />
                  <SwitchChainIcon
                    onClick={() => {
                      handleSwitchDropdowns()
                    }}
                  >
                    <SwitchIcon />
                  </SwitchChainIcon>
                  <Typography variant="body2">To</Typography>
                  <DropdownNetwork
                    items={switched ? NETWORK_L1_OPTIONS : NETWORK_L2_OPTIONS}
                    defaultItem={toNetwork}
                    onItemSelected={(option) => setToNetwork(option)}
                    error={false}
                    headers={[NETWORK_TYPE.MAINNET, NETWORK_TYPE.TESTNET]}
                  />
                </NetworkDropdowns>
                <FilterDropDown
                  items={FILTER_OPTIONS}
                  defaultItem={FILTER_OPTIONS[0]}
                  imgSrc={FilterIcon}
                  onItemSelected={(item) => {
                    setTransactionStatus(item.value)
                  }}
                  error={false}
                />
              </TableFilters>
            </div>
            <TableTransactionsContainer>
              <TransactionsTableHeader
                options={TableOptions}
              ></TransactionsTableHeader>
              {transactionsFound && (
                <TransactionsResolver
                  transactions={transactions}
                  transactionsFilter={{
                    fromNetworkChainId: fromNetwork.value as CHAIN_NAME,
                    toNetworkChainId: toNetwork.value as CHAIN_NAME,
                    status: transactionStatus as TRANSACTION_FILTER_STATUS,
                    targetHash: searchHistory,
                    startDate: filterStartDate,
                    endDate: filterEndDate,
                  }}
                ></TransactionsResolver>
              )}
            </TableTransactionsContainer>
          </Table>
        </>
      )}
      {!transactionsFound && (
        <NoHistory
          style={{ marginLeft: 'auto', marginRight: 'auto', padding: '20px' }}
        >
          <Svg src={noHistoryIcon} />
          <div>No Transactions Found.</div>
        </NoHistory>
      )}
      {!layer && (
        <NoHistory
          style={{ marginLeft: 'auto', marginRight: 'auto', padding: '20px' }}
        >
          <Svg src={noHistoryIcon} />
          <div>No History.</div>
          <Button
            onClick={() => dispatch(setConnect(true))}
            small
            label="Connect Wallet"
          />
        </NoHistory>
      )}
    </HistoryPageContainer>
  )
}

export default React.memo(History)
