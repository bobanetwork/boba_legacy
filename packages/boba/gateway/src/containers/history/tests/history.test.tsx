import { render, screen } from '@testing-library/react'
import React from 'react'
import CustomThemeProvider from 'themes'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import History from '../History'
import { TransactionsResolver } from '../TransactionsResolver'
import {
  CHAIN_NAME,
  ITransactionFilter,
  ITransactionsResolverProps,
  TRANSACTION_FILTER_STATUS,
} from '../types'
const sampleTransactions = require('./sampleTransactions.json')

const mockStore = configureStore()

const renderHistory = ({ options = null }: any) => {
  return render(
    <Provider
      store={mockStore({
        ui: {
          theme: 'dark',
        },
        transaction: sampleTransactions,
        setup: {
          accountEnabled: false,
          netLayer: true,
          baseEnabled: false,
          walletAddress: '0x1e2855A0EA33d5f293E5Ba2018874FAB9a7F05B3',
        },
        ...options,
      })}
    >
      <CustomThemeProvider>
        <History></History>
      </CustomThemeProvider>
    </Provider>
  )
}

const renderTransactionsResolver = (props: ITransactionsResolverProps) => {
  return render(
    <Provider
      store={mockStore({
        ui: {
          theme: 'dark',
        },
      })}
    >
      <CustomThemeProvider>
        <TransactionsResolver
          transactions={props.transactions}
          transactionsFilter={props.transactionsFilter}
        />
      </CustomThemeProvider>
    </Provider>
  )
}

describe('Testing history page', () => {
  beforeAll(() => {
    jest.useFakeTimers().setSystemTime(new Date(2023, 7, 24))
  })
  afterAll(() => {
    jest.useRealTimers()
  })

  const filter: ITransactionFilter = {
    fromNetworkChainId: CHAIN_NAME.Boba_Goerli,
    toNetworkChainId: CHAIN_NAME.Goerli,
    status: TRANSACTION_FILTER_STATUS.All,
    targetHash: '',
  }
  test('Test transaction resolver', () => {
    const { asFragment } = renderTransactionsResolver({
      transactions: sampleTransactions,
      transactionsFilter: filter,
    })
    expect(asFragment()).toMatchSnapshot()
  })
  test('Test History Page', () => {
    const { asFragment } = renderHistory({})
    expect(asFragment()).toMatchSnapshot()
  })
})
