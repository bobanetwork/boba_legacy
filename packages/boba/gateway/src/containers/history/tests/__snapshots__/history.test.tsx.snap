// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing history page Date Picker 1`] = `
<DocumentFragment>
  <div
    class="sc-cNNTdL XooGL"
  >
    <div
      class="sc-iWxGtA kjxmRV"
    >
      05/10/1970 12:00 am
    </div>
  </div>
</DocumentFragment>
`;

exports[`Testing history page Test History Page 1`] = `
<DocumentFragment>
  <div
    class="sc-eWhHU dDMUmU"
  >
    <div
      class="sc-brSOsn bhYExN"
    >
      <div
        class="sc-bxiBER mubpP"
      >
        <div>
          <div>
            <div
              data-src="magnifyingGlass.svg"
              src=""
            />
          </div>
        </div>
        <input
          class="sc-iHDLbo knCggh"
          placeholder="Search Here"
          value=""
        />
      </div>
      <div
        class="sc-dyTUbJ hUdBAf"
      >
        <p
          class="sc-duSInm bFEZbS sc-ckPVkh bOmlMV"
        >
          Date Range From
        </p>
        <div
          class="sc-cNNTdL iedVlo"
        >
          <div
            class="sc-iWxGtA cdSvk"
          >
            02/21/2023
          </div>
        </div>
        <p
          class="sc-duSInm bFEZbS sc-ckPVkh bOmlMV"
        >
          To
        </p>
        <div
          class="sc-cNNTdL iedVlo"
        >
          <div
            class="sc-iWxGtA cdSvk"
          >
            08/21/2023
          </div>
        </div>
      </div>
      <div
        class="sc-kJpdyg gLZWIR"
      >
        <p
          class="sc-duSInm bFEZbS sc-gAigJI lktCJV"
        >
          Date Range
        </p>
        <div
          class="sc-cNNTdL iedVlo"
        >
          <div
            class="sc-iWxGtA cdSvk"
          >
            02/21/2023 - 08/21/2023
          </div>
        </div>
      </div>
    </div>
    <div
      class="sc-hIiahf kVSiNY"
    >
      <div
        style="width: 100%; display: flex; flex-direction: column;"
      >
        <div
          class="sc-iCnXOK eRHODK"
          style="width: 100%;"
        >
          <div
            class="sc-kJwmPe ctcryJ"
          >
            <p
              class="sc-duSInm bFEZbS"
            >
              From
            </p>
            <div
              class="sc-fABZjn eemHrn dropdown sc-gCkVGe Mbgmo"
            >
              <div
                class="sc-gUMnWI dvASPu dropdown sc-gCkVGe Mbgmo"
              >
                <div
                  class="sc-ePzlA-D qoYai"
                >
                  <div
                    class="sc-hUOJWJ fJPHZV"
                  >
                    <div
                      class="sc-hjLWnZ dcydmI"
                    >
                      <div>
                        <div
                          data-src="allNetworks.svg"
                          src=""
                        />
                      </div>
                    </div>
                  </div>
                  <div>
                    All Networks
                  </div>
                  <div
                    class="sc-ccVCaX gXCdOf dropdown sc-gCkVGe Mbgmo"
                  >
                    <div>
                      <div
                        data-src="arrowdown.svg"
                        src=""
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="sc-pDgPE kbbFqH"
            >
              <div
                class="sc-hpGEOZ flecyt"
              >
                <div>
                  <div
                    data-src="switchIcon.svg"
                    src=""
                  />
                </div>
              </div>
            </div>
            <p
              class="sc-duSInm bFEZbS"
            >
              To
            </p>
            <div
              class="sc-fABZjn eemHrn dropdown sc-gCkVGe Mbgmo"
            >
              <div
                class="sc-gUMnWI dvASPu dropdown sc-gCkVGe Mbgmo"
              >
                <div
                  class="sc-ePzlA-D qoYai"
                >
                  <div
                    class="sc-hUOJWJ fJPHZV"
                  >
                    <div
                      class="sc-hjLWnZ dcydmI"
                    >
                      <div>
                        <div
                          data-src="allNetworks.svg"
                          src=""
                        />
                      </div>
                    </div>
                  </div>
                  <div>
                    All Networks
                  </div>
                  <div
                    class="sc-ccVCaX gXCdOf dropdown sc-gCkVGe Mbgmo"
                  >
                    <div>
                      <div
                        data-src="arrowdown.svg"
                        src=""
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="sc-ehYhRg jxAxWZ dropdown undefined"
          >
            <div
              class="sc-fia-DRh jfQAMp"
            >
              <div
                class="sc-hYqwXO ljMLMN"
              >
                <div
                  class="sc-lpcdUm hXxTAM"
                >
                  <div>
                    <div
                      data-src="filter.svg"
                      src=""
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-hCnrGf hXGTTH"
      >
        <div
          class="sc-gTRrQi jcFGvU sc-kcuKUB jWaUpl sc-kFDfur diVmzM"
        >
          <div
            class="sc-gTRrQi jcFGvU sc-kuWgmH iGlDXy sc-kFDfur diVmzM"
            style="max-width: 168px;"
          >
            <p
              class="sc-duSInm bFEZbS sc-jrrXlR iPvjOP"
            >
              Date
            </p>
          </div>
          <div
            class="sc-gTRrQi jcFGvU sc-kuWgmH iGlDXy sc-kFDfur diVmzM"
            style="max-width: 142px;"
          >
            <p
              class="sc-duSInm bFEZbS sc-jrrXlR iPvjOP"
            >
              From
            </p>
          </div>
          <div
            class="sc-gTRrQi jcFGvU sc-kuWgmH iGlDXy sc-kFDfur diVmzM"
            style="max-width: 142px;"
          >
            <p
              class="sc-duSInm bFEZbS sc-jrrXlR iPvjOP"
            >
              To
            </p>
          </div>
          <div
            class="sc-gTRrQi jcFGvU sc-kuWgmH iGlDXy sc-kFDfur diVmzM"
            style="max-width: 90px;"
          >
            <p
              class="sc-duSInm bFEZbS sc-jrrXlR iPvjOP"
            >
              Token
            </p>
          </div>
          <div
            class="sc-gTRrQi jcFGvU sc-kuWgmH iGlDXy sc-kFDfur diVmzM"
            style="max-width: 80px;"
          >
            <p
              class="sc-duSInm bFEZbS sc-jrrXlR iPvjOP"
            >
              Amount
            </p>
          </div>
          <div
            class="sc-gTRrQi jcFGvU sc-kuWgmH iGlDXy sc-kFDfur diVmzM"
            style="max-width: 88px;"
          >
            <p
              class="sc-duSInm bFEZbS sc-jrrXlR iPvjOP"
            >
              Status
            </p>
          </div>
        </div>
        <div
          class="sc-emqRaN cCQJsf"
        />
        <div
          class="sc-IeChK iNxsQx"
          style="margin-left: auto; margin-right: auto; padding: 20px;"
        >
          <div>
            <div>
              <div
                data-src="noHistory.svg"
                src=""
              />
            </div>
          </div>
          <div>
            No Transactions Matching Filter.
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Testing history page Test transaction resolver 1`] = `
<DocumentFragment>
  <div
    class="sc-emqRaN cCQJsf"
  />
  <div
    class="sc-IeChK iNxsQx"
    style="margin-left: auto; margin-right: auto; padding: 20px;"
  >
    <div>
      <div>
        <div
          data-src="noHistory.svg"
          src=""
        />
      </div>
    </div>
    <div>
      No Transactions Matching Filter.
    </div>
  </div>
</DocumentFragment>
`;
